oteapi_dlite/strategies/filter.py

```diff
@@ -0,0 +1,45 @@
+"""Trivial filter that adds an empty collection to the session."""
+# pylint: disable=no-self-use,unused-argument
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+import dlite
+from oteapi.models import SessionUpdate
+
+if TYPE_CHECKING:
+    from typing import Any, Dict, Optional
+
+    from oteapi.models import FilterConfig
+
+
+@dataclass
+class CreateCollectionStrategy:
+    """Trivial filter that adds an empty collection to the session.
+
+    **Registers strategies**:
+
+    - `("filterType", "create_collection")`
+
+    """
+
+    filter_config: "FilterConfig"
+
+    def initialize(
+        self, session: "Optional[Dict[str, Any]]" = None
+    ) -> "Dict[str, Any]":
+        """Initialize."""
+        if session is None:
+            raise ValueError("Missing session")
+        if "collection_id" in session:
+            raise KeyError("`collection_id` already exists in session.")
+        coll = dlite.Collection()
+
+        # Save reference to  the collection to ensure that it lives as long as
```

Comment:
```suggestion
        # Save reference to the collection to ensure that it lives as long as
```

Reply:
> I think you should create a models folder/module, under which you put a sessions.py file (or similar). 

I added a DLiteSessionUpdate in oteapi_dlite/models/session.py as suggested. I don't think we should create a new empty collection by default, because that will overwrite the existing collection in the session if a filter doesn't specify a default (another issue is that it will leak memory).

Maybe it is better to simply stick with SessionUpdate.

Reply:
> Note, if the collection_id is a UUID (as suggested by the property name used in the initialize() method) you could also use the uuid.UUID as a type or the specific pydantic type UUID4 (see the [docs](https://pydantic-docs.helpmanual.io/usage/types/#pydantic-types)).

The collection_id is an UUID, but we cannot just create an arbitrary UUID since it must correspond to an UUID of an existing collection.

Reply:
I wasn't suggesting to create an arbitrary UUID. I was suggesting to use a UUID type for the attribute for validation purposes.